src/extension.ts

```diff
@@ -208,6 +216,17 @@ namespace Configuration {
 }
 /* eslint-enable @typescript-eslint/no-namespace, no-inner-declarations */
 
+function setEnvironmentVariableContributions(ctx: vscode.ExtensionContext): void {
+    const settingValue: NodeJS.ProcessEnv = vscode.workspace.getConfiguration('docker').get<NodeJS.ProcessEnv>('environment', {});
+
+    ctx.environmentVariableCollection.clear();
+    ctx.environmentVariableCollection.persistent = true;
+
+    for (const key of Object.keys(settingValue)) {
+        ctx.environmentVariableCollection.replace(key, settingValue[key]);
```

Comment:
@Tyriar is this the right way to use this API? We call this `setEnvironmentVariableContributions` at extension activation and anytime the `docker.environment` setting is changed.